shared/js/advanced_telemetry_helper.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This module is a helper for recording telemetry data in Gecko’s telemetry
+ * system. Currently, only data from certified apps will be recorded.
+ *
+ * Usage Example:
+ *  var ath = new AdvancedTelemetryHelper();
+ *  // DO SOMETHING
+ *  // This creates and increments the counter on this metric called
+ *  // 'mycountmetric'
+ *  ath.count('mycountmetric'); // Count will be 1
+ *  ath.count('mycountmetric'); // Count is now 2
+ *  // DO SOMETHING
+ *  // This creates and logs a linear histogram called 'mylinearmetric' with
+ *  // value 100, minimum of 0, max of 1000, with 10 buckets.
+ *  ath.linear('mylinearmetric', 100, 0, 1000, 10);
```

Comment:
This is definitely subjective, but I think I agree with Russ. an extra line of code could make the encapsulation of each histogram a little more obvious, i.e:

```
var linear = ath.linear('mylinearmetric', 0, 1000, 10);
linear.log(100);
linear.log(800);

var exp = ath.exp('myexponentialmetric', 0, 10000, 10);
exp.log(10);
exp.log(8000);

```